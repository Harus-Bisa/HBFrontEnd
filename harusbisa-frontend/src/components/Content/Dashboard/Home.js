import React from "react";
import {connect} from "react-redux";

function Home(props){
    return(
        <div>
            <h1>{props.course.courseName}</h1>
            <p>{props.course.courseDescription}</p>
            <p>{props.course.startTerm} - {props.course.endTerm}</p>
            <p>Kode bergabung: {props.course.joinCode}</p>
            <p>Jumlah sesi: {props.course.numberOfLectures}</p>
            <p>Jumlah murid: {props.course.numberOfStudents}</p>
        </div>
    )
}

function mapStateToProps(state){
    return{
        course: state.course
    }
}

export default connect(mapStateToProps)(Home);