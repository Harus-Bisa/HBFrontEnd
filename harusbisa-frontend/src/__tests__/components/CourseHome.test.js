import React from "react";
import {shallow, mount} from "enzyme";
import configureMockStore from "redux-mock-store";
import '../../test-config';
import {Provider} from 'react-redux';
import Home from "../../components/Content/Dashboard/Home";
const mockStore = configureMockStore();
describe("Course Home component", () =>{
    it("matches snapshot", () =>{
        let initialState = {
            course:{
                "_id": "5db492c349a67b33b8d0a2a2",
                "course_name": "Calculus I",
                "course_description": "The basic of calculus (derivatives).",
                "start_term": "August 2019",
                "end_term": "December 2019",
                "number_of_lectures": 5,
                "instructor": "John Doe"
              }
        }
        let store = mockStore(initialState);
        let component = shallow(<Provider store={store}><Home/></Provider>)
        expect(component.html()).toMatchSnapshot()
    })
})