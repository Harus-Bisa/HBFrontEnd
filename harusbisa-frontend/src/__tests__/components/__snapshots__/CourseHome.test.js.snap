// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Home component matches snapshot 1`] = `"<div><h1></h1><p></p><p> - </p><p>Kode bergabung: </p><p>Jumlah sesi: </p><p>Jumlah murid: </p></div>"`;
